utils/spdx/src/main/kotlin/SpdxExpression.kt

```diff
@@ -477,7 +477,7 @@ class SpdxLicenseWithExceptionExpression(
 
         val spdxException = SpdxLicenseException.forId(exception)
         when (strictness) {
-            Strictness.ALLOW_ANY -> exception // Return something non-null.
+            Strictness.ALLOW_ANY -> Unit // Return something non-null to not trigger the fallback below.
```

Comment:
To be honest, I find the whole construct difficult to understand. Why is an exception constructed first, and then another one is thrown? I would rather expect something like

```
val spdxException = when(strictness) {
  ...
}
spdxException?.let { throw it }
```

Reply:
Note that `spdxException` is not a JVM exception, but an SPDX license exception instance. I'll rename the variable to `spdxLicenseException` to make that clearer.

Reply:
I decided to rewrite this completely, getting rid of the `when`.